engine/univSubst.ml

```diff
@@ -131,9 +131,9 @@ let nf_evars_and_universes_opt_subst f subst =
   let rec aux c =
     match kind c with
     | Evar (evk, args) ->
-      let args = List.map aux args in
-      (match try f (evk, args) with Not_found -> None with
-      | None -> mkEvar (evk, args)
+      let args' = List.Smart.map aux args in
+      (match try f (evk, args') with Not_found -> None with
+      | None -> if args == args' then c else mkEvar (evk, args')
```

Comment:
This change is not in the bench, will you rerun?

Reply:
Bench rerunning here: https://ci.inria.fr/coq/view/benchmarking/job/benchmark-part-of-the-branch/954.